src/util/DataUtils.ts

```diff
@@ -12,7 +12,7 @@ export const mathSign = (value: number) => {
 };
 
 export const isPercent = (value: string | number): value is `${number}%` =>
-  _.isString(value) && value.indexOf('%') === value.length - 1;
+  _.isString(value) && value.length > 1 && value.indexOf('%') === value.length - 1;
```

Comment:
I do believe that many of these changes will actually fix bugs in a few edge cases, and I believe them to be valuable, but I would prefer to see smaller PRs, each with more focus. 

The focus being generateCategoricalChart means it is almost the whole library.